pycbc/io/hdf.py

```diff
@@ -799,6 +799,21 @@ def get_end_time(self):
             ref_times = self.get_coincfile_array('time1')
         return ref_times
 
+    def get_ifos(self):
+        ifo_list = []
+        N_trigs = len(self.get_coincfile_array('template_id'))
+        try:  # First try new-style format
+            ifos = self.coinc_file.h5file.attrs['ifos'].split(' ')
+            for ifo in ifos:
+                ifo_trigs = np.where(self.get_coincfile_array(ifo+'/time') < 0,
+                                     '-', ifo)
+                ifo_list.append(ifo_trigs)
+            ifo_list = np.array([','.join(trig[trig != '-']) \
+                          for trig in iter(np.array(ifo_list).T)])
+        except KeyError:  # Else fall back on old two-det format
+            ifo_list = np.tile('H1,L1', N_trigs)
```

Comment:
I suggest adding an explicit comment here saying that you *assume* the analysis involved Hanford and Livingston. In principle one could have analyzed GEO+Virgo, and the result of this method would then be incorrect.

\begin{thinking_loudly}
What if the workflow tagged *every* output HDF file with an attribute reporting the metadata about which detectors were analysed?
\end{thinking_loudly}